### ISSUE TITLE
Docs: Add method definitions to component API docs

### ISSUE THREAD

**Issue by FlipWarthog** [primefaces/primevue#4702]
### Describe the feature you would like to see added

Component methods are not current documented in the API docs, e.g., `show` and `hide` methods of AutoComplete can only be known by looking at the PrimeVue source: https://primevue.org/autocomplete/#api.autocomplete

Both PrimeFaces and PrimeReact document such methods:
[PrimeFaces example](https://primefaces.github.io/primefaces/jsdocs/classes/src_PrimeFaces.PrimeFaces.widget.AutoComplete-1.html)
[PrimeReact example](https://primereact.org/autocomplete/#api.AutoComplete.methods)

This was discussed in the PrimeLand discord; I'm not sure if it's a difference in best practices between the frameworks/libraries or something that should be added.

### Is your feature request related to a problem?

Some solutions to community problems have involved using methods not documented in the component API.

**Comment by BenJackGill** [comment]
I agree. The AutoComplete docs have `show` and `hide` in the [emits section](https://v3.primevue.org/autocomplete/#api.autocomplete.emits.show) of the docs. This is correct since those are events are emitted when the suggestions are shown or hidden, but I also needed a way to programatically `hide` the suggestions and I didn't realise there was also a useful `hide` **method** that I could call until stumbling upon this issue.

I'm sure other components need the docs updated also. As I can't remember seeing a Methods section in any of those docs. Quasar documents their component methods well and I would love to see the same here in PrimeVue.
